backends/build_system/awscli_venv.py

```diff
@@ -0,0 +1,91 @@
+import os
+import subprocess
+import site
+import sys
+import venv
+import pathlib
+
+from constants import (
+    BOOTSTRAP_REQUIREMENTS_LOCK,
+    BUILD_REQUIREMENTS_LOCK,
+    ROOT_DIR,
+    IS_WINDOWS,
+    BIN_DIRNAME,
+    PYTHON_EXE_NAME,
+)
+from utils import Utils
+
+
+class AwsCliVenv:
+    _PARENT_SCRIPTS_TO_COPY = [
+        "pyinstaller",
+    ]
+
+    def __init__(self, venv_dir: str, utils: Utils = None):
+        self._venv_dir = venv_dir
+        if utils is None:
+            utils = Utils()
+        self._utils = utils
+
+    def create(self):
+        self._utils.create_venv(self._venv_dir, with_pip=True)
+
+    def bootstrap(self, download_deps=False):
+        if download_deps:
+            self._install_requirements(BOOTSTRAP_REQUIREMENTS_LOCK)
+            self._install_requirements(BUILD_REQUIREMENTS_LOCK)
+        else:
+            self._copy_parent_packages()
+        self._install_awscli()
+
+    def _copy_parent_packages(self):
+        for site_package in site.getsitepackages():
+            self._utils.copy_directory_contents_into(
+                site_package, self._site_packages()
+            )
+        parent_scripts = pathlib.Path(sys.executable).parents[0]
+        for script in self._PARENT_SCRIPTS_TO_COPY:
+            source = os.path.join(parent_scripts, script)
+            if self._utils.path_exists(source):
+                self._utils.copy_file(source, os.path.join(self.bin_dir, script))
+
+    def _install_requirements(self, requirements_file, cwd=None):
+        self._pip_install(["-r", requirements_file], cwd=cwd)
+
+    def _install_awscli(self):
+        self._pip_install(
+            [
+                ROOT_DIR,
+                "--no-build-isolation",
+                "--no-cache-dir",
```

Comment:
Was `--use-feature=in-tree-build` removed? I'm assuming it is because it is the default now? It might be worth updating the proposal to remove that flag as well to reflect the implementation now.

Reply:
Yep this is the default now, the flag was removed from pip.